Checks the deployment of 2 partition and 1 consumer (kafka,partition)

gicappa opened the pull request:
The CI is now testing that when helm performs an installation in the case of Kafka Partitioned it is possible to have a number of consumer different (<=) from the number of the partitions (i.e. partitions=2, consumers=1)

ref. Activiti/Activiti#3698

gicappa committed:
Activiti/Activiti#3698 Fake commit to trigger the deployment

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

    In the case of kafka partitioned leg of the matrix we need to add a new
    consumer to test 2 consumers to fetch from 2 partitions testing the
    autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 Testing 2 query consumers in kafka partitioned

In the case of kafka partitioned leg of the matrix we need to add a new
consumer to test 2 consumers to fetch from 2 partitions testing the
autoRebalanceEnabled=true flag.

gicappa committed:
Activiti/Activiti#3698 FIX: flawky test ProcessInstanceNotifications

The test
org.activiti.cloud.qa.story.ProcessInstanceNotifications.verifyAllNotificationsAreReceived
is failing once out of three times in the CI environment but not
locally.
Increasing session and submission timeouts to understand if they are too
strict.

gicappa committed:
Activiti/Activiti#3698 FIX: flawky test ProcessInstanceNotifications

The test
org.activiti.cloud.qa.story.ProcessInstanceNotifications.verifyAllNotificationsAreRecei
is failing once out of three times in the CI environment but not
locally.
Increasing session and submission timeouts to understand if they are too
strict.

gicappa committed:
Activiti/Activiti#3698 FIX: flawky test ProcessInstanceNotifications

The test
org.activiti.cloud.qa.story.ProcessInstanceNotifications.verifyAllNotificationsAre
is failing once out of three times in the CI environment but not
locally.
Increasing session and submission timeouts to understand if they are too
strict.

gicappa committed:
Activiti/Activiti#3698 FIX: flawky test ProcessInstanceNotifications

The test
org.activiti.cloud.qa.story.ProcessInstanceNotifications.verifyAllNotificationsAre
is failing once out of three times in the CI environment but not
locally.
Launching the runtime bundle related tests once to understand if the
issue is related to the missing reentrancy of the test

erdemedeiros commented on .github/workflows/main.yml:
```diff
@@ -229,38 +229,130 @@ jobs:
+          # Showing the deployment result
           kubectl get all -n $PREVIEW_NAME
 
+          # Launching the tests
           make test/modeling-acceptance-tests
+
+      - name: Lanunch runtime acceptance test (not kafka partitioned)
+        if: ${{ matrix.messaging-broker != 'kafka' && matrix.messaging-partitioned != 'partitioned' }}
+        run: |
+          ### Variables ###
+          export SSO_PROTOCOL=https
+          export GATEWAY_PROTOCOL=https
+          export GLOBAL_GATEWAY_DOMAIN=$CLUSTER_NAME.$CLUSTER_DOMAIN
+          export GATEWAY_HOST=gateway-$PREVIEW_NAME.$GLOBAL_GATEWAY_DOMAIN
+          export SSO_HOST=identity-$PREVIEW_NAME.$GLOBAL_GATEWAY_DOMAIN
+
+          make test/runtime-acceptance-tests
+
+      - name: Increase the consumers number and launch runtime acceptance test (only kafka partitioned)
+        if: ${{ matrix.messaging-broker == 'kafka' && matrix.messaging-partitioned == 'partitioned' }}
```
With the current configuration, we are running only the modeling tests for `rabbitmq-partitioned` and `kafka-non-partiotioned`. We need to have runtime acceptance tests running for those configurations as well.

gicappa replied:
Talking with @erdemedeiros we decided to modify the subject of the test of the code. So this comment is not applicable anymore.

erdemedeiros reviewed (changes requested):
There is a problem with the current configuration. Check my comment below. Could you update the title of the PR as well?

gicappa committed:
Activiti/Activiti#3698 New test of partitions and consumers numbers

Added a test to check if the deployment has created a topic with two
partitions and one consumer as the customer was asking.

erdemedeiros committed:
Merge branch 'develop' into GH-3698-kafka-consumer-dynamic-scaling-preview

erdemedeiros reviewed (approved):
LGTM!
